src/controller/CHIPDevice.cpp

```diff
@@ -84,60 +124,26 @@ CHIP_ERROR Device::SendMessage(System::PacketBufferHandle buffer, PayloadHeader
     // Try session resumption if needed
     if (err != CHIP_NO_ERROR && !resend.IsNull() && mState == ConnectionState::SecureConnected)
     {
+        Messaging::SendFlags sendFlags;
+
         mState = ConnectionState::NotConnected;
 
         ReturnErrorOnFailure(LoadSecureSessionParameters(ResetTransport::kYes));
 
-        err = mSessionManager->SendMessage(mSecureSession, std::move(resend));
+        sendFlags.Set(Messaging::SendMessageFlags::kFromInitiator, true).Set(Messaging::SendMessageFlags::kNoAutoRequestAck, true);
```

Comment:
Why are we setting these flags on _resend_ but not on initial send?  As above in chip-send, I don't think we should set these at all.

Reply:
All messages carried over TempZCL protocol should have kFromInitiator and kNoAutoRequestAck flags set as I explained above.

Reply:
This explanation should be in the code.